test/evergreen.yml

```diff
@@ -2640,7 +2587,59 @@ buildvariants:
       LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libeatmydata.so PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
   tasks:
     - name: ".pull_request !.windows_only !.pull_request_compilers !.python"
-    - examples-c-test
+    - name: make-check-test
+    - name: examples-c-test
+
+- name: ubuntu1804-msan
+  display_name: "! Ubuntu 18.04 MSAN"
```

Comment:
Do we want it to be marked as a required builder? Depending on how long the tasks run and how useful they could capture issues, we can consider adding certain tasks in this builder into PR testing.

Reply:
I'm running a new patch build with my changes. Let's see what the runtimes for the new variants look like.

Reply:
How's the new patch build going?

Reply:
The makespan was 42 minutes for the three variants.

Reply:
That's a bit too long. I think our current PR runtime is within 30mins. If deciding to exclude the MSAN & UBSAN tests from PR testing, I'd suggest using the combined `make-check-test` task to replace individual tasks in the 2 new builders, to save some AWS cost (only need to spin up one EC2 instead of multiple).

Reply:
I will second not going past 30 mins for PR testing.

Reply:
@lukech I can't use `make-check-test` as we need to bypass one of the csuite tests that hangs on a `cursor.modify` call when run with msan/ubsan.

Reply:
I see. We had a similar issue previously that certain test is not appropriate to run for ASAN build. The solution at the time is to bypass the test in the test source code (an [example](https://github.com/wiredtiger/wiredtiger/blob/develop/test/csuite/wt4156_metadata_salvage/main.c#L446-L448)). Do you think we can simulate it by creating a MSAN/UBSAN flag?

Reply:
I was able to resolve this by adding `-O1` to the compiler options.